UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/NMDataService.cs

```diff
@@ -346,33 +347,60 @@ public async Task<byte[]> DownloadFSSZipFileAsync(string batchId, string fileNam
 
         }
 
-        public async Task<List<ShowFilesResponseModel>> GetCumulativeBatchFiles(string correlationId)
+        public async Task<ShowNMFilesResponseModel> GetCumulativeBatchFiles(string correlationId)
         {
             try
             {
-                _logger.LogInformation(EventIds.GetCumulativeFilesResponseStarted.ToEventId(), "Maritime safety information request to get cumulative NM files response started with _X-Correlation-ID:{correlationId}", correlationId);
+                BatchSearchResponse searchResult = new();
+                bool isCached = false;
+                const string frequency = "Cumulative";
+                const string partitionKey = "Public";
+                const string rowKey = "CumulativeKey";
 
-                string accessToken = await _authFssTokenProvider.GenerateADAccessToken(_userService.IsDistributorUser, correlationId);
+                if (_cacheConfiguration.Value.IsFssCacheEnabled)
+                {
+                    BatchSearchResponseModel batchSearchResponseModel = await _fileShareServiceCache.GetBatchResponseFromCache(partitionKey, rowKey, frequency, correlationId);
 
-                const string searchText = $" and $batch(Frequency) eq 'cumulative'";
+                    if (batchSearchResponseModel.BatchSearchResponse != null)
+                    {
+                        searchResult = batchSearchResponseModel.BatchSearchResponse;
+                        isCached = true;
+                    }
+                }
 
-                IFileShareApiClient fileShareApiClient = new FileShareApiClient(_httpClientFactory, _fileShareServiceConfig.Value.BaseUrl, accessToken);
+                if (searchResult.Entries == null)
+                {
+                    _logger.LogInformation(EventIds.GetCumulativeFilesResponseStarted.ToEventId(), "Maritime safety information request to get cumulative NM files response started with _X-Correlation-ID:{correlationId}", correlationId);
 
-                IResult<BatchSearchResponse> result = await _fileShareService.FSSBatchSearchAsync(searchText, accessToken, correlationId, fileShareApiClient);
+                    string accessToken = await _authFssTokenProvider.GenerateADAccessToken(_userService.IsDistributorUser, correlationId);
 
-                BatchSearchResponse searchResult = result.Data;
+                    const string searchText = $" and $batch(Frequency) eq 'cumulative'";
```

Comment:
Better to use the `const` to reduce magic strings please:
```suggestion
                    const string searchText = $" and $batch(Frequency) eq '{frequency}'";
```

Reply:
Sorry, But I have not understood.

1. We have already used const.
2. We can't use {frequency}, as this function is not dynamic.

Reply:
Changes are done as required.